libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/util/CacheResultUtils.kt

```diff
@@ -51,6 +53,106 @@ internal object CacheResultUtils {
         }
     }
 
+    data class CacheResultKeyRouteTraffic<R>(
+        val route: NavigationRoute,
+        val trafficProvider: (RouteLeg) -> List<String>?
+    ) :
+        CacheResultCall<(NavigationRoute, (RouteLeg) -> List<String>?) -> R, R> {
+
+        override fun invoke(f: (NavigationRoute, (RouteLeg) -> List<String>?) -> R) = f(
+            route,
+            trafficProvider
+        )
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as CacheResultKeyRouteTraffic<*>
+
+            if (route.id != other.route.id) return false
+            if (trafficProvider != trafficProvider) return false
+            if (
+                route.directionsRoute.legs()?.size !=
+                other.route.directionsRoute.legs()?.size
+            ) {
+                return false
+            }
+            if (!trafficIsEqual(route, other.route)) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = route.id.hashCode()
+            result = 31 * result + trafficProvider.hashCode()
+            route.directionsRoute.legs()?.forEach { routeLeg ->
+                result = 31 * result + routeLeg.annotation()?.congestion().hashCode()
+                result = 31 * result + routeLeg.annotation()?.congestionNumeric().hashCode()
+            }
+            return result
+        }
+
+        private fun trafficIsEqual(route: NavigationRoute, other: NavigationRoute): Boolean {
+            val hasNumericTraffic: Boolean = route.directionsRoute.routeOptions()?.annotationsList()
+                ?.contains(DirectionsCriteria.ANNOTATION_CONGESTION_NUMERIC) ?: false
+            val sameTrafficType = hasNumericTraffic ==
```

Comment:
If 
```
route.directionsRoute.routeOptions()?.annotationsList()
                ?.contains(DirectionsCriteria.ANNOTATION_CONGESTION_NUMERIC)
``` 
is null and 
```
other.directionsRoute.routeOptions()?.annotationsList()
                    ?.contains(DirectionsCriteria.ANNOTATION_CONGESTION_NUMERIC)
```
is null, sameTrafficType will be false. I don't think it's expected.

Reply:
Adjusted.